### DISCUSSION TITLE
presence of theta in the consistent tanegnt operator

### DISCUSSION THREAD

**Discussion by GhiathMonnet** [thelfer/tfel#543] [General]
Dear ALL, 
If I understand correctly the implicit scheme, the theta parameter is used inside the @Integraor block to integrate the internal variables over the increment deto. When checking the formulas of the consistent tangent operator Dt in several mfront files, I found that theta exists also in the expression of Dt. I beleive that the expression of Dt is sent to the calling solver which, in principal, knows nothing about theta, since theta is a variable of the internal Newton method. Does this make any sens for you ? Does mfront send the value of theta to the solver ?
Many thanks.
Ghiath

**Comment by thelfer** [comment]
Consistent in "consistent tangent operator" means consistent with the numerical algorithm.

`theta` is indeed internal but has an influence on the computed stress, thus it  appears in the expression of the consistent tangent operator.

See page 24 of this document for an example of the computation of the consistent tangent operator: https://thelfer.github.io/tfel/web/documents/mfront/behaviours.pdf

**Comment by GhiathMonnet** [comment]
OK I see. The tangent is calculated at the middle of the step.
Many thanks.
Cheers.
Ghiath :+1:

**Reply by thelfer** [reply]
I think that you misunderstood.
It is computed by taken into account the numerical algorithm. It is not the value at the middle of the time step.
